0000-ldap-sync-tool.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: `mgr-ldapsync`: LDAP sync tool
+- Start Date: 2019, Oct 28
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The `mgr-ldapsync` is a tool that performs synchronisation of the
+users between Uyuni server and LDAP.
+
+# Motivation
+[motivation]: #motivation
+
+Uyuni server can authenticate users from LDAP via PAM. For this, users
+needs to be added to Uyuni with the flag "PAM authentication" and thus
+their credentials will be verified via LDAP directory. Synchronisation
+tool should transparently synchronise users from certain groups and/or
+roles in LDAP and reflect that in Uyuni server, overriding local
+changes.
+
+Often LDAP configuration at infrastructures is unpredictable. This
+tool should bring certain boundaries and over up most common LDAP
+setups for users and their roles management.
+
+# Detailed design
+[design]: #detailed-design
+
+The tool should work as an aggregator, collecting users from LDAP and
+Uyuni server synchronising them.
+
+## Functionality Aspects
+
+The following functions should be implemented:
+
+- Distinguish between local users in Uyuni and LDAP users.
+- Prevent UID clashes between the local users and LDAP users.
+- Prevent accidental modification or removal of the local user.
+- Flexible support various LDAP schemas, such as eDirectory, POSIX,
+  Active Directory etc.
+- Role management.
+- Dry run mode (preview only).
+
+## Configuration
+
+Configuration should be default in a separate file
+`/etc/rhn/ldapsync.conf`. Alternatively, user should be able to
```

Comment:
Why, oh why, we keep using /etc/rhn and not something like /etc/uyuni ? :-(

Reply:
Well, everybody knows this directory. I would not change it as long as we change it for all the config files. This is of cause a bigger task which could break a lot of things.

We would need to dedicate some time to such a change and we need to do it on a new major version. It would also need a migration old-path to new path.

Reply:
> Well, everybody knows this directory. I would not change it as long as we change it for all the config files. This is of cause a bigger task which could break a lot of things.
> 
> We would need to dedicate some time to such a change and we need to do it on a new major version. It would also need a migration old-path to new path.

I know, I know. It was a rhetorical question :-(

Reply:
@paususe because:
1. It has historical reasons
2. Everyone knows it and very likely hard-coded it in their own scripts (we can symlink it though for backward compatibilities)
3. It is already there, so why I would add another directory? If we migrate it one day, then we migrate it all together